widget/select.go

```diff
@@ -200,7 +207,10 @@ func (s *Select) CreateRenderer() fyne.WidgetRenderer {
 	text.Alignment = fyne.TextAlignLeading
 
 	objects := []fyne.CanvasObject{text, icon}
-	return &selectRenderer{widget.NewShadowingRenderer(objects, widget.ButtonLevel), icon, text, s}
+	r := &selectRenderer{widget.NewShadowingRenderer(objects, widget.ButtonLevel), icon, text, s}
+	r.updateLabel()
+	r.updateIcon()
```

Comment:
Refresh locks before the helpers and unlocks after, rather than locking within each helper. The only other place these helpers are used are in CreateRenderer which doesn't need the lock as no one has the reference to the renderer yet

Reply:
The acquired lock does not belong to the renderer but to the widget. That’s why I think it is acquired because the updates might change the widget.
Yes, I think this is questionable. Feel free to improve.

Reply:
Oh actually this is incorrect, the lock should be held here as it is the widget's lock, not the widgetrenderer's lock.

Will fix in next commit.